hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java

```diff
@@ -927,6 +923,44 @@ private static File urlToFile(@Nonnull URL url) {
         return resourceConfigs;
     }
 
+    /**
+     * Registers given serializer for a given class for the scope of the job.
+     * Both will be accessible to all the code attached to the underlying
+     * pipeline or DAG, but not to any other code. (An important example is
+     * the {@code IMap} data source, which can instantiate only the classes
+     * from the Jet instance's classpath.)
+     *
+     * Serializer must have no-arg constructor.
+     *
+     * @implNote Backing storage for this method is an {@link IMap} with a
+     * default backup count of 1. When adding big files as a resource, size
+     * the cluster accordingly in terms of memory, since each file will have 2
+     * copies inside the cluster(primary + backup replica).
+     *
+     * @return {@code this} instance for fluent API
+     */
+    @Nonnull
+    public <T, S extends Serializer> JobConfig addSerializer(Class<T> clazz, Class<S> serializerClass) {
+        addClass(clazz);
```

Comment:
We should not add classes. We should just register the serializer and the user is responsible for having the classes on classpath. There are other ways to make the class available at runtime, this is the least common one.